jscomp/build_tests/super_errors/fixtures/component_missing_prop_children.res

```diff
@@ -0,0 +1,7 @@
+// Since the React transform isn't active in the tests, mimic what the transform outputs.
+module Component = {
+  @res.jsxComponentProps
+  type props<'name> = {name: 'name}
+
+  let make = (): props<'name> => {children: ""}
```

Comment:
I don't get this. Is this really what the transform would output?

Why is the `make` function _returning_ props? And not `React.element`?

Reply:
No it's not what the transform would output, but it doesn't matter, what matters is that a record of the given type is being created, which is what the transform would do. If we want to make this clearer we could set up a test suite with the transform enabled. But I don't know whether that's cumbersome to do. Do you know?

Reply:
@zth #7048

Reply:
@cknitt the problem now is that React itself isn't included. There's a `react_ppx` folder that does similar tests _with_ the transform and React copied into the project itself. We could mimic that, but that's a larger change.

Reply:
@zth I don't think you need React, you can use the `Jsx` module that comes with the compiler.

E.g. something like

```rescript
@jsx.component
let make = (~name) => Jsx.string(name)
```

should work fine.

Reply:
Ok, well, you don't need it in my example, but you are right, you need the externals for the JSX runtime if you want to create a React element.

Reply:
Yup.

Reply:
@cknitt let's just worry about the actual error message in this PR, and then we can figure out how to set up the React transform for this separately.